pkg/api/kptfile/v1alpha2/validation.go

```diff
@@ -46,24 +46,42 @@ func (p *Pipeline) validate() error {
 	fns = append(fns, p.Validators...)
 	for i := range fns {
 		f := fns[i]
-		err := f.validate()
+		// fnIdx is the index of this function in corresponding function
+		// type
+		fnIdx := i
+		// fnType is the type of this function
+		fnType := "mutators"
+		if i >= len(p.Mutators) {
+			// this function is a validator
+			fnIdx = i - len(p.Mutators)
+			fnType = "validators"
+		}
```

Comment:
It might be better to have two loops for each `mutators` and `validators` to have simple code here.

Reply:
done